extmod/modbluetooth.c

```diff
@@ -155,13 +148,13 @@ STATIC void bluetooth_uuid_print(const mp_print_t *print, mp_obj_t self_in, mp_p
     mp_obj_bluetooth_uuid_t *self = MP_OBJ_TO_PTR(self_in);
 
     if (self->type == MP_BLUETOOTH_UUID_TYPE_16) {
-        mp_printf(print, "UUID16(0x%04x)", self->uuid._16);
+        mp_printf(print, "UUID16(0x%02x%02x)", self->data[1], self->data[0]);
     } else if (self->type == MP_BLUETOOTH_UUID_TYPE_32) {
-        mp_printf(print, "UUID32(0x%08x)", self->uuid._32);
+        mp_printf(print, "UUID32(0x%02x%02x%02x%02x)", self->data[3], self->data[2], self->data[1], self->data[0]);
     } else if (self->type == MP_BLUETOOTH_UUID_TYPE_128) {
         mp_printf(print, "UUID128('");
         for (int i = 0; i < 16; ++i) {
-            mp_printf(print, "%02x", self->uuid._128[15-i]);
+            mp_printf(print, "%02x", self->data[15-i]);
```

Comment:
Could the printing be combined into one, with the first bit like:

    mp_printf(print, "UUID%u(%s", self->type * 8, self->type <=4 ? "0x" : "'");